lib/fake_artist/table.ex

```diff
@@ -11,35 +15,81 @@ defmodule FakeArtist.Table do
   end
 
-  def add_self(table_pid) do
-    GenServer.call(table_pid, :add_self)
+  def add_self(pid) do
+    GenServer.call(pid, :add_self)
+  end
+
+  def start_game(pid) do
+    GenServer.call(pid, :start_game)
+  end
+
+  def progress_game(pid) do
+    GenServer.call(pid, :progress_game)
+  end
+
+  def choose_category(pid) do
+    GenServer.call(pid, :choose_category)
   end
 
   # Server Callbacks
   def init([name]) do
-    {:ok, {%{}, name, 0}}
+    {:ok, {%{}, name, 0, nil}}
   end
 
-  def handle_call({:update_name_tag, {id, name_tag}}, _from, {id_map, table_name, user_count}) do
+  def handle_call(
+        {:update_name_tag, {id, name_tag}},
+        _from,
+        {id_map, table_name, user_count, state_pid}
+      ) do
     id_map = Map.put(id_map, id, name_tag)
 
-    FakeArtistWeb.Endpoint.broadcast("table:#{table_name}", "update", %{
-      names: Map.values(id_map)
-    })
+    FakeArtistWeb.Endpoint.broadcast("table:#{table_name}", "update", %{names: Map.values(id_map)})
 
-    {:reply, :ok, {id_map, table_name, user_count}}
+    {:reply, :ok, {id_map, table_name, user_count, state_pid}}
   end
 
-  def handle_call(:add_self, {from_pid, _}, {id_map, table_name, user_count}) do
+  def handle_call(:add_self, {from_pid, _}, {id_map, table_name, user_count, state_pid}) do
     Logger.info(fn -> "started monitoring #{inspect(from_pid)}" end)
     Process.monitor(from_pid)
     Logger.info(fn -> "player count increased from #{user_count} to #{user_count + 1}" end)
-    {:reply, :ok, {id_map, table_name, user_count + 1}}
+    {:reply, :ok, {id_map, table_name, user_count + 1, state_pid}}
+  end
+
+  def handle_call(:start_game, _from, {id_map, table_name, user_count, _old_state_pid}) do
+    {:ok, state_pid} = Map.keys(id_map) |> FakeArtist.Game.start_link()
+    %{seats: seats, active_seat: active_seat} = FakeArtist.Game.get_state(state_pid)
+    players = assemble_players(seats, id_map, active_seat)
+
+    FakeArtistWeb.Endpoint.broadcast("table:#{table_name}", "update_game", %{players: players})
+
+    {:reply, :ok, {id_map, table_name, user_count, state_pid}}
   end
 
-  def handle_info({:DOWN, _ref, :process, _from, _reason}, {id_map, table_name, user_count}) do
+  def handle_call(:progress_game, _from, {id_map, table_name, user_count, state_pid}) do
+    %{seats: seats, active_seat: active_seat} = FakeArtist.Game.progress_game(state_pid)
+    players = assemble_players(seats, id_map, active_seat)
+
+    FakeArtistWeb.Endpoint.broadcast("table:#{table_name}", "update_game", %{players: players})
+
+    {:reply, :ok, {id_map, table_name, user_count, state_pid}}
+  end
+
+  def handle_call(:choose_category, _from, {id_map, table_name, user_count, state_pid}) do
+    %{seats: seats, active_seat: active_seat} = FakeArtist.Game.choose_category(state_pid)
+    players = assemble_players(seats, id_map, active_seat)
+
+    FakeArtistWeb.Endpoint.broadcast("table:#{table_name}", "update_game", %{players: players})
```

Comment:
might as well make the `assemble_players/3` fn just return the `%{players: players}` map directly, right?

Reply:
I'd rather not. I feel like the handle_call function is assembling the pieces, and then they're being put into the "state" that is passed around in that .broadcast function. It feels right to me.

Later on we might want to have a second element in the map, and we could plug it in right there.

Reply:
that's fine!